causal_testing/data_collection/data_collector.py

```diff
@@ -61,62 +120,22 @@ def run_system_with_input_configuration(
 
 
 class ObservationalDataCollector(DataCollector):
-    def __init__(self, scenario: Scenario):
-        super().__init__()
-        self.scenario = scenario
 
-    def collect_data(self, csv_path: str) -> pd.DataFrame:
+    def __init__(self, scenario: Scenario, csv_path: str):
+        super().__init__(scenario)
+        self.csv_path = csv_path
+
+    def collect_data(self, **kwargs) -> pd.DataFrame:
         """
         Read a csv containing execution data for the system-under-test into a pandas dataframe and filter to remove any
-        data which does is invalid for the scenario-under-test. Data is invalid if it does not meet the constraints
+        data which is invalid for the scenario-under-test. Data is invalid if it does not meet the constraints
         outlined in the scenario-under-test (Scenario).
+
+        :param scenario: Scenario for which the observational data is collected.
         :param csv_path: Path to the csv containing execution data.
         :return scenario_execution_data_df: A pandas dataframe containing execution data that is valid for the
         scenario-under-test.
         """
-        execution_data_df = pd.read_csv(csv_path)
+        execution_data_df = pd.read_csv(self.csv_path, **kwargs)
```

Comment:
I am not sure this is a good way to use **kwargs. If you are using the **kwargs specifically to pass args to read_csv, it might be better to have them as explicit keyword arguments.

Reply:
Fair enough. I was mainly just keen to avoid duplicating all the kwargs to `pd.read_csv`. The only one I actually needed was `index_col`, so I just added support for that and removed `**kwargs`.

Reply:
Update: Turns out codacity didn't like that, so I had to revert back :|